### DISCUSSION TITLE
I have an issue with activity time line

### DISCUSSION THREAD

**Discussion by Crocky44** [deejay-hub/timeline-lwc#168] [Q&A] [answered]
I activated Activity history on Timeline. It works well except for the biggest account where I have have such error....
"Cannot read properties of undefined (reading 'replace')"

![image](https://user-images.githubusercontent.com/95232865/153401130-6889d389-03af-4411-9f02-c476dc036cea.png)

Could you please help me? Do you have an idea?

**Comment by deejay-hub** [comment]
Hi Crocky44. We would need to replicate the issue. The best way for us to do this is for you to share as much information as possible.

- what version of time warp do you have installed?
- what parent object are you using (sounds like Account)?
- can you share screen shots of all active timeline configuration records for the parent object you're having issues with?
- what is different about the Account you're having issues with? Is it the number of child records? If so could you query the database to let us know how many records you have for each child object?
- Can you identify the child record that causes the issue by inactivating all child records in the custom metadata type then activating them one at a time?
- do you use Person Accounts?
- what time range are you plotting? You can go back and forward up to 10 years. What values are set for you?

Is that something you'd be willing to share with us?

**Comment by deejay-hub** [comment]
@Crocky44 actually looks like we might have an explanation. If you've activated 'OpenActivities' or 'ActivityHistory' custom metadata types then we can replicate the issue when the field value returned from the date is null. 

You can help us confirm this is the case by opening the 2 metadata types corresponding to 'OpenActivities' AND 'ActivityHistory' and change the default value for Position Date field from ActivityDate (which is the Due Date) to a CreatedDate.

<img width="1059" alt="image" src="https://user-images.githubusercontent.com/7021883/153534488-8ecc42fc-4744-4a00-a9c1-faa6032de105.png">

This should resolve the issue since this value is never NULL. **If you can confirm this resolves the issue we can raise a defect to fix it.**

In the meantime the only workaround is to inactivate OpenActivities and ActivityHistory metadata types and activate Tasks and Events instead.

**Comment by Crocky44** [comment]
You are right, with created date it is working. perfectly  The point is that for this account, I cannot see NULL (Empty) value for ActivityDate (I made a report but nothing wrong appeared) . Is there a way I can detect it in order I delete this "No good" Task?

Thanks for all

++

**Comment by deejay-hub** [comment, marked as answer]
@Crocky44 thanks for testing this. This is a defect - the Time Warp component should not error is the ActivityDate field is blank for ActivityHistory and OpenActivities. We have raised a [defect](https://github.com/deejay-hub/timeline-lwc/issues/169) for this and it is already fixed and ready for the next release (scheduled for May 22).

In the meantime we do have a workaround.
1. Create a new field on the Activity object. Type: **Formula** Formula Return Type: **Date/Time**.
2. Add the following formula 'IF ( ISNULL(ActivityDate) || ISBLANK(ActivityDate) , DATETIMEVALUE('1900-01-01 12:00:00'), DATETIMEVALUE(ActivityDate) )'
<img width="1396" alt="image" src="https://user-images.githubusercontent.com/7021883/153972601-17b483f7-7942-4e8c-a996-fa5d7aeb4da7.png">

3. Navigate to Setup->Custom Metadata Types and click **Manage Records** next to 'Timeline Configuration'
4. Click **Edit** on the metadata corresponding to 'Account_ActivityHistory' and modify the field 'Position Date Field' to specify the API Name of the field you created in Step 1 and 2
<img width="1381" alt="image" src="https://user-images.githubusercontent.com/7021883/153972706-1cb621c8-ce59-4597-8852-3fb8f0dbe9e5.png">
5. Repeat for the metadata corresponding to Account_OpenActivities.


Let us know if this solves your issue. This will ignore any blank or null dates by substituting a value in the past but always use the Due Date if the value is populated.

**Comment by deejay-hub** [comment]
Hi @Crocky44 it's been a few weeks without an update. We'll keep this open for 2 more weeks but we're hoping you've managed to use the workaround above. Any update for us?

**Comment by Crocky44** [comment]
Hi, 
Yes, problem is solved. Indeed one task had no date…I delete the record and it is working perfectly now.

Thanks

Ben

**Comment by deejay-hub** [comment]
Good news. Thanks for the update.

**Comment by deejay-hub** [comment]
Fixed in v1.11
